src/main/java/org/jabref/logic/preferences/CustomApiKeyPreferences.java

```diff
@@ -0,0 +1,34 @@
+package org.jabref.logic.preferences;
+
+public class CustomApiKeyPreferences {
+    private final String name;
+    private boolean useCustom;
+    private String defaultApiKey;
+
+    public CustomApiKeyPreferences(String name, boolean useCustom, String defaultApiKey) {
```

Comment:
Why would you keep a defaultApiKey in the preferences? Just throw the logic about defaultApiKey and useCustom away.
Just check in the fetcher, if there is something stored in the preferences and if so, use it. Otherwise use the build-in key in the build info.

Reply:
The use custom key stuff would then just be something for the ui in the preferences dialog. The checkbox is activated, if there is a key stored in the preferences. If not, its unchecked. If the preferences are stored, the preferences dialog should check, if the checkbox is checked for use custom, if yes, it should store the user defined key. If not, the key is being thrown away.

Reply:
I think you could even go one step further and remove the checkbox. If something is put in the text field it is used as a custom api key.

Reply:
The user can choose not to use the custom API key temporarily, but save it in JabRef, and only need to check the checkbox when reusing it next time.

Reply:
Ok, seems fair. Keep the use custom preference.